tests/components/gdacs/test_sensor.py

```diff
@@ -0,0 +1,100 @@
+"""The tests for the GDACS Feed integration."""
+from asynctest import CoroutineMock, patch
+
+from homeassistant.components import gdacs
+from homeassistant.components.gdacs import DEFAULT_SCAN_INTERVAL
+from homeassistant.components.gdacs.sensor import (
+    ATTR_CREATED,
+    ATTR_LAST_UPDATE,
+    ATTR_LAST_UPDATE_SUCCESSFUL,
+    ATTR_REMOVED,
+    ATTR_STATUS,
+    ATTR_UPDATED,
+)
+from homeassistant.const import (
+    ATTR_ICON,
+    ATTR_UNIT_OF_MEASUREMENT,
+    CONF_RADIUS,
+    EVENT_HOMEASSISTANT_START,
+)
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from tests.common import async_fire_time_changed
+from tests.components.gdacs import _generate_mock_feed_entry
+
+CONFIG = {gdacs.DOMAIN: {CONF_RADIUS: 200}}
+
+
+async def test_setup(hass):
+    """Test the general setup of the integration."""
+    # Set up some mock feed entries for this test.
+    mock_entry_1 = _generate_mock_feed_entry(
+        "1234", "Title 1", 15.5, (38.0, -3.0), attribution="Attribution 1",
+    )
+    mock_entry_2 = _generate_mock_feed_entry("2345", "Title 2", 20.5, (38.1, -3.1),)
+    mock_entry_3 = _generate_mock_feed_entry("3456", "Title 3", 25.5, (38.2, -3.2),)
+    mock_entry_4 = _generate_mock_feed_entry("4567", "Title 4", 12.5, (38.3, -3.3))
+
+    # Patching 'utcnow' to gain more control over the timed update.
+    utcnow = dt_util.utcnow()
+    with patch("homeassistant.util.dt.utcnow", return_value=utcnow), patch(
+        "aio_georss_client.feed.GeoRssFeed.update", new_callable=CoroutineMock
```

Comment:
`asynctest.patch` will use a CoroutineMock by default if the target is a coroutine.

Reply:
OK